src/bin/addr2line.rs

```diff
@@ -208,6 +222,20 @@ fn main() {
             }
         }
 
+        // If --section is given, add the section address to probe.
+        let probe = probe.map(|probe| {
+            if let Some(section_range) = section_range {
+                if probe < (section_range.end - section_range.begin) {
+                    probe + section_range.begin
+                } else {
+                    // If addr >= section size, treat it as if no line number information was found.
+                    0
```

Comment:
On this code path, the result should be that `probe` is assigned `None`, rather than `Some(0)`. So might need to use `and_then` instead of `map`.

Reply:
Resolved in 376f95ac1f26ba097b9c47e64fe602c1eeec3116